repos/spack_repo/builtin/packages/aocl_libmem/package.py

```diff
@@ -47,22 +48,58 @@ class AoclLibmem(CMakePackage):
     variant(
         "vectorization",
         default="auto",
+        when="@:5.0",
         description="Use hardware vectorization support",
         values=("avx2", "avx512", "auto"),
         multi=False,
     )
 
+    # Add new variant for dynamic dispatcher support
+    variant(
+        "dynamic-dispatch",
+        default=False,
+        when="@5.1:",
+        description="Single portable optimized library"
+        " to execute on different x86 CPU architectures",
+    )
+    variant(
+        "vectorization",
+        default="none",
```

Comment:
Note that for a while this would not actually give the correct default value since two identical variants for different when version ranges with different defaults deferred to the first default in the concretizer. @alalazo Is this still the case?